### PR TITLE
Commerce

### PR DESCRIPTION
**Pull request by vicb**

### CODE REVIEW

File: examples/vercel-commerce/lib/shopify/index.ts [hunk 1]
```diff
@@ -205,6 +205,10 @@ const reshapeProducts = (products: ShopifyProduct[]) => {
 };
 
 export async function createCart(): Promise<Cart> {
+  if (!process.env.SHOPIFY_STORE_DOMAIN) {
```
Comment (vicb) [inline examples/vercel-commerce/lib/shopify/index.ts:208]: I think we should avoid modifying the app more than what is needed to run on workers.
If this is to help we the tests, we can skip them when there is an error message.

File: examples/vercel-commerce/e2e/base.spec.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  await expect(page.getByRole('link', { name: 'Created by ▲ Vercel' })).toBeVisible();
+});
+
+test('the actual commerce functionality', async ({ page }) => {
+  test.skip(!process.env.SHOPIFY_STORE_DOMAIN, 'Populate the .dev.vars file to run the test');
+
+  await page.goto('/');
+
+  const hanraShirtLink = page.getByRole('link', { name: 'Hanra Shirt Hanra Shirt $108.' });
```
Comment (vicb) [inline examples/vercel-commerce/e2e/base.spec.ts:21]: This test is brittle as it depends on the content of the shop.
At least we should test what shop we are serving but even then it makes little sense if we do not control the content of that shop.

**Comment by vicb** [comment]
@dario-piotrowicz thanks for fixing the PR.

LGTM with a few comments:
- I wouldn't update the app more than what is needed to run on workers
- I wouldn't test things depending on the shop content that we don't control

File: examples/vercel-commerce/e2e/base.spec.ts (hunk 2 above)
Comment (dario-piotrowicz) [reply examples/vercel-commerce/e2e/base.spec.ts:21]: yes I totally agree that this is quite brittle, I just didn't want to modify the app more than necessary so, with the html in place there I could not find something better to test this functionality

I am happy to remove this or modify the app to be more easily testable, whatever you prefer 👍

File: examples/vercel-commerce/lib/shopify/index.ts (hunk 1 above)
Comment (dario-piotrowicz) [reply examples/vercel-commerce/lib/shopify/index.ts:208]: based on this and your other comment I can remove all the modification I did just to enable testing the commerce functionality

File: examples/vercel-commerce/e2e/base.spec.ts (hunk 2 above)
Comment (dario-piotrowicz) [reply examples/vercel-commerce/e2e/base.spec.ts]: oh... I just realized, I could very easily mock the network requests.... 🤔 🤷

File: examples/vercel-commerce/lib/shopify/index.ts (hunk 1 above)
Comment (dario-piotrowicz) [reply examples/vercel-commerce/lib/shopify/index.ts]: removed

File: examples/vercel-commerce/e2e/base.spec.ts (hunk 2 above)
Comment (vicb) [reply examples/vercel-commerce/e2e/base.spec.ts]: IMO a basic test as the first one is enough for now

**Comment by dario-piotrowicz** [comment]
@vicb the tweaks I made to the app were actually necessary to get any e2e to work

so I've removed the e2es altogether... let me know what you prefer, we can either keep the e2es but re-add some slight tweaks to the app or have the app without any sort of e2e...

**Comment by vicb** [comment]
@dario-piotrowicz it's becoming hard to review, could you squash your commits in fewer commits to ease the review.

> the tweaks I made to the app were actually necessary to get any e2e to work

I'm not sure to understand the root cause here.
If the CI creates the `.dev.vars` then it should be able to bring up the app?
But I guess it's not a big deal anyway.

**Comment by dario-piotrowicz** [comment]
> @dario-piotrowicz it's becoming hard to review, could you squash your commits in fewer commits to ease the review.

sure

> > the tweaks I made to the app were actually necessary to get any e2e to work
> 
> I'm not sure to understand the root cause here.
> If the CI creates the .dev.vars then it should be able to bring up the app?

I've remove that part, I think that we do not want the e2es to work only when the `.dev.vars` file is fully populated because this would be problematic for external contributors that would be forced to get a shopify key to get the e2es to work (adding a significant friction point in my opinion)

And the `wrangler dev` command without all the `.dev.vars` although works if you view the app with your browser is not getting detected as working properly by playwright (I suspect that playwright is getting some 500s when trying to see if the server is running)

**Comment by dario-piotrowicz** [comment]
@vicb I've rebased and squashed my commits, please have another look 🙏

**Comment by vicb** [comment]
Looks good, thanks for making this work!

**Pull request merged**